crates/build-utils/src/content_reader.rs

```diff
@@ -71,4 +75,31 @@ impl ContentReader {
         // Return a clone of the block
         Ok(first_block.clone())
     }
+
+    pub fn transaction(test_case: &Value, block: &Value) -> Result<Value, eyre::Error> {
```

Comment:
From what I could see we only use `ContentReader::transaction` in order to get the private key. Could we perhaps just move this logic into the private key function? We can then move the whole private key logic in `format_test_content`, pass the block and get the pvt key using the static only if needed